### PR TITLE
Fixed false positive nested-min-max for nested lists

### LINKED ISSUE #9307: False positive `nested-min-max` for nested lists
### Bug description

```python
"""False positive `nested-min-max` for nested lists"""
a = [[1, 2, 3], [4, 5, 6]]
print(max(max(a)))  # Output: 6
print(max(*a))  # Output: [4, 5, 6]
```

### Configuration

_No response_

### Command used

```shell
pylint a.py
```

### Pylint output

```python
************* Module a
a.py:3:6: W3301: Do not use nested call of 'max'; it's possible to do 'max(*a)' instead (nested-min-max)
```

### Expected behavior

Pylint gives wrong advice. If `a` is a nest list, `max(max(a))` is not equivalent  `max(*a)`.

### Pylint version

```shell
pylint 3.0.3
astroid 3.0.2
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
```

### OS / Environment

_No response_

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by udifuchs**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Nesting is useful for finding the maximum in a matrix.
Therefore, pylint now allows nesting of the form:
```
max(max([[1, 2, 3], [4, 5, 6]]))
```

Closes #9307

### CODE REVIEW

File: tests/functional/n/nested_min_max.py [hunk 1]
```diff
@@ -54,3 +54,11 @@
 
 max(3, max(list(range(4))))  # [nested-min-max]
 max(3, *list(range(4)))
+
+# Nesting is useful for finding the maximum in a matrix
+# No message if extrenal call has exactly 1 argument
```
Comment (jacobtylerwalls) [inline tests/functional/n/nested_min_max.py:59]: Hey! Hope to find time for a review soon. In the meantime, a small typo:
```suggestion
# No message if external call has exactly 1 argument
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
LGTM, great fix ! I'd like to see the primer's result before merging,, I think it depends on 
#9330 being merged and then a rebase on it.

**Comment by Pierre-Sassoulas** [comment]
It can be rebased now :)

**Review by Pierre-Sassoulas — APPROVED** [review]
Good job fixing the primers @jacobtylerwalls :)

**Pull request merged**